src/Coordination/KeeperStateMachine.cpp

```diff
@@ -182,6 +196,12 @@ void KeeperStateMachine::commit_config(const uint64_t /* log_idx */, nuraft::ptr
     cluster_config = ClusterConfig::deserialize(*tmp);
 }
 
+void KeeperStateMachine::rollback(uint64_t log_idx, nuraft::buffer & /*data*/)
+{
+    std::lock_guard lock(storage_and_responses_lock);
+    storage->rollbackRequest(log_idx);
```

Comment:
What is the purpose of this now?

Reply:
`rollback` undos `pre_commit`. If leaders change, it's normal for it overwrite some logs. 
`pre_commit` is called instantly after a log is appended to the LogStore so we need to somehow undo it if that log is overwritten.